src/common/notifications.cpp

```diff
@@ -0,0 +1,84 @@
+#include "pch.h"
+#include "notifications.h"
+
+#include <unknwn.h>
+#include <winrt/base.h>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Data.Xml.Dom.h>
+#include <winrt/Windows.Foundation.Collections.h>
+#include <winrt/Windows.UI.Notifications.h>
+#include <winrt/Windows.ApplicationModel.Background.h>
+
+using namespace winrt::Windows::ApplicationModel::Background;
+using winrt::Windows::Data::Xml::Dom::XmlDocument;
+using winrt::Windows::UI::Notifications::ToastNotification;
+using winrt::Windows::UI::Notifications::ToastNotificationManager;
+
+namespace
+{
+    constexpr std::wstring_view TASK_NAME = L"PowerToysBackgroundNotificationsHandler";
+    constexpr std::wstring_view TASK_ENTRYPOINT = L"PowerToysNotifications.BackgroundHandler";
+}
+
+void notif::register_background_toast_handler()
+{
+    // Re-request access to clean up from our previous versions
+    BackgroundExecutionManager::RemoveAccess();
+    BackgroundExecutionManager::RequestAccessAsync().get();
+
+    BackgroundTaskBuilder builder;
+    ToastNotificationActionTrigger trigger{ L"PowerToys" };
+    builder.SetTrigger(trigger);
+    builder.TaskEntryPoint(TASK_ENTRYPOINT);
+    builder.Name(TASK_NAME);
+
+    const auto tasks = BackgroundTaskRegistration::AllTasks();
+    const bool already_registered = std::any_of(begin(tasks), end(tasks), [=](const auto& task) {
+        return task.Value().Name() == TASK_NAME;
+    });
+    if (already_registered)
+    {
+        return;
+    }
+    try
+    {
+        (void)builder.Register();
+    }
+    catch (...)
+    {
+        // Couldn't register the task, nothing we can do
+    }
+}
+
+void notif::show_toast(std::wstring_view message)
+{
+    show_toast_background_activated(L"Unknown", message, {});
+}
+
+void notif::show_toast_background_activated(std::wstring_view background_handler_id, std::wstring_view message, std::vector<std::wstring_view> button_labels)
```

Comment:
Maybe we should escape the strings? The `message` can be put in `<![CDATA[ ... ]]>`, for `button_labels` we could just change all `"` to`&quot;`. Sooner or later someone will try to have a button with a quote.

Reply:
@bzoz I was thinking of adding show_toast_xml if someone wants a full customization in another PR. These functions are specifically for plain-text. Maybe the argument could be better named...

Reply:
I'll check if the strings are not automatically escaped